src/layout/Flex.js

```diff
@@ -0,0 +1,240 @@
+    typeof value === 'string' &&
+    ['around', 'between', 'evenly'].indexOf(value) !== -1
+  ) {
+    return `space-${value}`;
+  } else {
+    return getAlignment(value);
+  }
+};
+
+const styles = {
+  root: ({
+    breakpoints,
+    alignItems,
+    direction,
+    gutterWidth,
+    inline,
+    margin,
+    marginEnd,
+    marginHorizontal,
+    marginLeft: propMarginLeft,
+    marginRight: propMarginRight,
+    marginStart,
+    justifyContent,
+    theme,
+    wrap
+  }) => {
+    const rtl = theme.direction === 'rtl';
+    const gutter = getGutterSize(theme, gutterWidth);
+    const marginLeft =
+      propMarginLeft ||
+      (rtl ? marginEnd : marginStart) ||
+      marginHorizontal ||
+      margin;
+    const marginRight =
+      propMarginRight ||
+      (rtl ? marginStart : marginEnd) ||
+      marginHorizontal ||
+      margin;
+
+    const mapValueToProperty = (property, value) => {
+      const map = {
+        alignItems: getAlignment,
+        display: value =>
+          value === undefined || value === false ? 'flex' : 'inline-flex',
+        flexDirection: value => value,
+        flexWrap: value =>
+          value ? 'wrap' : value === false ? 'nowrap' : value,
+        marginLeft: value => getMargin(gutter, theme, value),
+        marginRight: value => getMargin(gutter, theme, value),
+        justifyContent: getJustification
+      };
+
+      return map[property](value);
+    };
+
+    return getResponsiveStyles({
+      breakpoints,
+      mapValueToProperty,
+      styles: {
+        alignItems,
+        display: inline,
+        flexDirection: direction,
+        flexWrap: wrap,
+        marginLeft,
+        marginRight,
+        justifyContent
+      },
+      theme
+    });
+  }
+};
+
+const Root = createStyledComponent(Box, styles.root, {
+  displayName: 'Flex',
+  filterProps: ['inline']
+});
+
+const ThemedRoot = withTheme(
+  ({ breakpoints, children, gutterWidth, theme, ...restProps }) => {
+    const rootProps = {
+      breakpoints,
+      gutterWidth,
+      ...restProps
+    };
+
+    let flexItems;
+    flexItems = Children.map(children, child => {
+      const {
+        breakpoints: propBreakpoints,
+        margin: propMargin,
+        marginHorizontal,
+        marginLeft,
+        marginRight
+      } = child.props;
+      let props = {
+        breakpoints: propBreakpoints || breakpoints
+      };
+
+      if (gutterWidth) {
+        const margin =
+          marginHorizontal || propMargin || getGutterSize(theme, gutterWidth);
+        props = {
+          ...props,
+          marginLeft: marginLeft || margin,
+          marginRight: marginRight || margin
+        };
+      }
+      return cloneElement(child, props);
+    });
+
+    return <Root {...rootProps}>{flexItems}</Root>;
+  }
+);
+
+/**
+ * Flex component is used together with [FlexItem](../flex-item) to lay out
+ * other components in a flexible, and optionally responsive, manner.
+ */
+export default function Flex({
+  // $FlowFixMe
```

Comment:
Hmmm..... We should try to look into these `$FlowFixMes` a bit more.

Reply:
*sigh* On it...

Reply:
Looks like tweaking defaultProps will fix it.